backend/shigai-serve/src/main.rs

```diff
@@ -0,0 +1,127 @@
+use std::io::prelude::{Read, Write};
+use std::net::TcpListener;
+use std::net::TcpStream;
+
+const SRC_ROOT: &str = "../../frontend/src";
+
+fn main() {
+    let listener = TcpListener::bind("127.0.0.1:7878").unwrap();
+
+    for stream in listener.incoming() {
+        let stream = stream.unwrap();
+
+        handle_connection(stream);
+    }
+}
+
+fn handle_connection(mut stream: TcpStream) {
+    let mut buffer = [0; 2048];
+
+    if let Ok(buf_length) = stream.read(&mut buffer) {
+        let message = String::from_utf8_lossy(&buffer[..buf_length]).to_string();
+        println!("{}", message);
+
+        let http_req: http::HTTPRequest = http::process_request(message);
+        router::route(http_req, stream);
+    } else {
+        let response = "HTTP/1.1 400 Bad Request \r\n\r\n";
```

Comment:
I think we shouldn't use carriage returns. Can't we use a library for this? Oh, well. I guess it's okay for now.